### PR TITLE
[SwiftLint] Add note for params files for `path` option

### PR DESCRIPTION
**Pull request by suetsuguk**
This PR contains below issues:

- Add support for params files for file path arguments #499

### CODE REVIEW

**Comment by ybiquitous** [comment]
Can we add just a link to the SwiftLint document?

**Comment by suetsuguk** [comment]
As for "params files", I cannot find any document (even in the command-line help) but the [PR](https://github.com/realm/SwiftLint/pull/3313).

**Comment by ybiquitous** [comment]
Can you please copy-and-paste the command-line help here?

**Comment by suetsuguk** [comment]
Here is the help message:

```console
analyzer_runner@fc1f02cc55bf:~$ swiftlint version
0.40.2
analyzer_runner@fc1f02cc55bf:~$ swiftlint help lint
Print lint warnings and errors (default command)

[--path (string)]
        the path to the file or directory to lint

[--use-stdin]
        lint standard input

[--config (string)]
        the path to SwiftLint's configuration file

[--strict]
        fail on warnings

[--lenient]
        downgrades serious violations to warnings, warning threshold is disabled

[--force-exclude]
        exclude files in config `excluded` even if their paths are explicitly specified

[--use-script-input-files]
        read SCRIPT_INPUT_FILE* environment variables as files

[--benchmark]
        save benchmarks to benchmark_files.txt and benchmark_rules.txt

[--reporter (string)]
        the reporter used to log errors and warnings

[--quiet]
        don't print status logs like 'Linting <file>' & 'Done linting'

[--cache-path (string)]
        the directory of the cache used when linting

[--no-cache]
        ignore cache when linting

[--enable-all-rules]
        run all rules, even opt-in and disabled ones, ignoring `whitelist_rules`

[[""]]
        list of paths to the files or directories to lint
analyzer_runner@fc1f02cc55bf:~$
```

**Comment by ybiquitous** [comment]
Um..., I understand this behavior is undocumented anywhere. 😓 

I'm a bit worried about whether we could maintain this description if this undocumented behavior would be changed.
But surely, it might be useful for some users if the description would be present.

OK, let's go on.

File: docs/tools/swift/swiftlint.md [hunk 1]
```diff
@@ -48,6 +48,16 @@ You can use several options to fine-tune SwiftLint to your project.
...
 This option allows you to specify a file or directory to analyze.
 
+This also allows you to specify a special format `"@path/to/file"` (concatenation of `@` and a file name). You can create a file which contains a list of file names to be analyzed as shown below and pass it to the analyzer with this option.
+
+```text
+test1.swift
+test2.swift
+foo/test3.swift
+```
+
```
Comment (ybiquitous) [inline docs/tools/swift/swiftlint.md:58]: [suggestion] How about adding a concrete example like this?

---

For example, when you have the file `config/swiftlint_target.txt` with the following content,

```text
test1.swift
test2.swift
foo/test3.swift
```

then `sider.yml` should be below:

```yaml
linter:
  swiftlint:
    path: "@config/swiftlint_target.txt"
```

**Review by ybiquitous — COMMENTED** [review]
[suggestion] Please forget updating `0.40.0` to `0.40.2` in "Supported Version".

Also, this pull request should be merged after Sider will provide this feature actually.

File: docs/tools/swift/swiftlint.md (hunk 1 above)
Comment (suetsuguk) [reply docs/tools/swift/swiftlint.md:60]: Fixed. commit 3787542050fd55eb015785b0f6cfb1306b66dfb0

**Comment by suetsuguk** [comment]
> [suggestion] Please forget updating `0.40.0` to `0.40.2` in "Supported Version".

I will change the supported version along with other tools at a different PR where I will update `2020.md`.

**Comment by ybiquitous** [comment]
OK. I've changed the PR title to avoid confusion.

**Review by ybiquitous — APPROVED** [review]

File: docs/tools/swift/swiftlint.md [hunk 2]
```diff
@@ -48,6 +48,26 @@ You can use several options to fine-tune SwiftLint to your project.
 
 This option allows you to specify a file or directory to analyze.
 
+This also allows you to specify a special format `"@path/to/file"` (concatenation of `@` and a file name). You can create a file which contains a list of file names to be analyzed and pass it to the analyzer with this option.
```
Comment (yykamei) [inline docs/tools/swift/swiftlint.md:51]: How about using the word "prefix"? The relevant code uses `hasPrefix` internally.

https://github.com/realm/SwiftLint/blob/6d2e8cfc061ecfd73d711b7f607a1dc4bb2b0bda/Source/swiftlint/Helpers/LintableFilesVisitor.swift#L33


```suggestion
This also allows you to specify a special format `"@path/to/file"` (prefixed with `@`). You can create a file which contains a list of file names to be analyzed and pass it to the analyzer with this option.
```

**Review by yykamei — APPROVED** [review]

File: docs/tools/swift/swiftlint.md (hunk 2 above)
Comment (suetsuguk) [reply docs/tools/swift/swiftlint.md:51]: I was wondering the same thing. But a filename prefixed with @ such as "@files.lst" leads to misunderstanding that users should make a name of "@files.lst" not "files.lst".

Comment (yykamei) [reply docs/tools/swift/swiftlint.md:51]: Do you mean users may write like this "config/@swiftlint_target.txt"?

Comment (suetsuguk) [reply docs/tools/swift/swiftlint.md:51]: Now that we have a concrete example, we can avoid misunderstanding. I will apply your suggestion.

Comment (suetsuguk) [reply docs/tools/swift/swiftlint.md]: Fixed. commit 36bdba3d4a58b423817e31b841380199dde97092

**Review by yykamei — APPROVED** [review]
Thank you

**Pull request merged**